src/dev_guides/build.md

```diff
@@ -18,7 +18,20 @@
 # How to build Parsec
 
 This project is coded in the Rust Programming Language. To build it, you first need to [install Rust](https://www.rust-lang.org/tools/install).
-To build and run the service, execute `cargo run`. `parsec` will then wait for clients.
+
+Because the providers supported by Parsec are dependent on libraries and/or hardware features present on the platform, the build is fragmented through Rust features so that the resulting binary only contains the desired providers. Currently the service provides the following features: `mbed-crypto-provider`, `pkcs11-provider`, and `tpm-provider`.
+
+In order for the service to be spun up, a number of parameters are required in a TOML file. The repository contains an [example](https://github.com/parallaxsecond/parsec/blob/master/config.toml) of such a configuration file which shows all the options and their default values.
+
+To build and run the service, set `DESIRED_FEATURES` to be a subset of the features mentioned above, space or comma separated, and execute:
+```bash
+cargo run --features $DESIRED_FEATURES
```

Comment:
Should we add `--no-default-features` to that command? Or mention the default ones?

Reply:
Unfortunately, yeah, forgot!